Brain/brain_utils.h

```diff
@@ -87,22 +87,48 @@ String make_packet(struct sensorInfo sensor) {
  * Populated the fields of the valve and returns the action to be taken
  */
 int decode_received_packet(String packet, valveInfo *valve) {
+  Serial.println(packet);
   int data_start_index = packet.indexOf(',');
+//  Serial.print("start ind: ");
+//  Serial.println(data_start_index);
+  if(data_start_index == -1){
+    return -1;
+  }
   int valve_id = packet.substring(1,data_start_index).toInt();
   const int data_end_index = packet.indexOf('|');
+//  Serial.print("end ind: ");
+//  Serial.println(data_end_index);
+  if(data_end_index == -1){
+    return -1;
+  }
   int action = packet.substring(data_start_index + 1,data_end_index).toInt();
   
   String checksumstr = packet.substring(data_end_index + 1, packet.length()-2);
-//  char checksum_char[5];
-//  checksumstr.toCharArray(checksum_char, 5);
-  char *checksum_char = checksumstr.c_str();
+  Serial.println(checksumstr);
+  char checksum_char[5];
+  checksumstr.toCharArray(checksum_char, 5);
+//  char *checksum_char = checksumstr.c_str();
   uint16_t checksum = strtol(checksum_char, NULL, 16);
+  Serial.print("check: ");
+  Serial.println(checksum);
   
-  char const *data = packet.substring(1,data_end_index).c_str();
-  int count = data_end_index - 1; // sanity check; is this right? off by 1 error?
-  uint16_t check = Fletcher16((uint8_t *) data, count);
-  if (check == checksum) {
+  const int count = packet.substring(1, data_end_index).length(); // sanity check; is this right? off by 1 error?
+  Serial.println(count);
+  char data[count+1];// = packet.substring(1,data_end_index).c_str();
+  packet.substring(1, data_end_index).toCharArray(data, count + 1);
```

Comment:
Just out of curiosity, was there something buggy with the previous way of getting the data?

Reply:
it was doing some weird conversion. I think it had something to do with encoding, or maybe we were doing something incorrectly, lemme add it to trello as a task, someone should figure out why it wasn't working.

Reply:
Ok once I get the I2C working I can look into this also